### DISCUSSION TITLE
Moving ZST39 and beginner controller questions

### DISCUSSION THREAD

**Discussion by greenbus4** [zwave-js/zwave-js-ui#4070] [Q&A]
Sorry, this is long, but not very complex questions. Any help is appreciated.

I have a few beginner question. I've had my Z-wave network running pretty well for over a year. About 50 nodes. But, I've been nervous about my controller failing w/o a clear understanding how to deal with that.

Background

Yesterday I set up a Zen05 out in the yard and it's just barely working do to its range. I decided to try Long Range and so moved from [Z-wave JS to Z-wave JS UI](https://community.home-assistant.io/t/switching-z-wave-js-addons-with-minimal-downtime-z-wave-js-official-to-z-wave-js-ui-community/409904 ) to include the switch using LR. 

Including the Zen05 for LR worked fine, but moving it outside it didn't work at all. My controller is in the center of the house so there's three wood-framed walls just to get outside, then another 75 feet to the Zen05 switch.

Home Assistant is running on a Rpi 4b and I have a [USB powered hub](https://www.amazon.com/dp/B00TPMEOYM?ref_=ppx_hzsearch_conn_dt_b_fed_asin_title_2&th=1) and a [USB extension cable](https://www.amazon.com/dp/B002KKZAIU?ref_=ppx_hzsearch_conn_dt_b_fed_asin_title_2&th=1).

I want to relocate the HA server & controller closer to the outside switch to use Long Range, but I want to power it all over PoE+, so I'd have to get rid of the powered USB hub.  i.e. Plug the ZST39 and extension directly into the Raspberry Pi.

Questions:

1) My serial port is configured with the "by-id" path (/dev/serial/by-id/usb-Zooz_800_Z-Wave_Stick_533D004242-if00). Am I correct that path stays the same even if I get rid of the USB hub?

2) My ZST39 is running FW: v1.1 & SDK: v7.18.1. I cannot do an NVM backup because [I need 7.19.0 or higher](https://github.com/zwave-js/node-zwave-js/issues/5257) to do the backup. I'm unaware of any issues with my controller, so I'm hesitant to blindly upgrade the ZST39 w/o having a backup. A bit of a catch-22 as I'd need to upgrade first to be able to do the backup.

Should I try and use [Simplicity Studio](https://www.silabs.com/developer-tools/simplicity-studio) and do the NVM backup there first?

3) And if Simplicity Studio is the right approach, I assume that I can unplug the Controller from the Raspberry Pi and plug the ZST39 into my Mac safely. Is that correct?

4) I've considered replacing my controller with something like [TubesZB](https://tubeszb.com/product/z-wave-poe-kit/). Can you do NVM backup and then restore to a different device? I cannot imagine re-including all my devices and then dealing with broken automations in HA for devices ids that changed and or different entity ids.

Only "guide" I found for this was [this video](https://www.youtube.com/watch?v=lktIFOcg7GI) that uses Simplicity Studio. Is there a good guide on how to replace a controller without having to rebuild everything?

Finally, on a somewhat funny note: I had strategically placed a few repeater switches to extend my mesh to reach more distant nodes. I even considered buying a dedicated range extender. Well, using Z-wave JS UI I now can see the mesh layout and NONE of those switches are being used by those distant nodes as I had assumed. I have a Zen16 maybe 10 feet in clear air from one of my switches and it is instead connecting to a node 50 feet away. So odd.

I've rebuilt the routes many times and they still take what seem to me as nonsensical routes to the controller. I have nodes that crisscross the house on their hops before they get to the controller. Should I just trust the rebuilding to get it right? 

Thanks. Especially if you made it this far...

**Comment by robertsLando** [comment]
> My serial port is configured with the "by-id" path (/dev/serial/by-id/usb-Zooz_800_Z-Wave_Stick_533D004242-if00). Am I correct that path stays the same even if I get rid of the USB hub?

Yes as the serial id is unique of your device, it remains always the same the hub doesn't matter (I use a one too and it doesn't change the id with or without it)

> My ZST39 is running FW: v1.1 & SDK: v7.18.1. I cannot do an NVM backup because https://github.com/zwave-js/node-zwave-js/issues/5257 to do the backup. I'm unaware of any issues with my controller, so I'm hesitant to blindly upgrade the ZST39 w/o having a backup. A bit of a catch-22 as I'd need to upgrade first to be able to do the backup.

> Should I try and use [Simplicity Studio](https://www.silabs.com/developer-tools/simplicity-studio) and do the NVM backup there first?

> And if Simplicity Studio is the right approach, I assume that I can unplug the Controller from the Raspberry Pi and plug the ZST39 into my Mac safely. Is that correct?

The best one to answer this is @AlCalzone. 

> I've considered replacing my controller with something like [TubesZB](https://tubeszb.com/product/z-wave-poe-kit/). Can you do NVM backup and then restore to a different device? I cannot imagine re-including all my devices and then dealing with broken automations in HA for devices ids that changed and or different entity ids.

That's possible but not from/to all different kind of controllers. Again @AlCalzone is the best to answer this :) 

> Finally, on a somewhat funny note: I had strategically placed a few repeater switches to extend my mesh to reach more distant nodes. I even considered buying a dedicated range extender. Well, using Z-wave JS UI I now can see the mesh layout and NONE of those switches are being used by those distant nodes as I had assumed. I have a Zen16 maybe 10 feet in clear air from one of my switches and it is instead connecting to a node 50 feet away. So odd.

> I've rebuilt the routes many times and they still take what seem to me as nonsensical routes to the controller. I have nodes that crisscross the house on their hops before they get to the controller. Should I just trust the rebuilding to get it right?

From Mesh UI you can click on a specific node and from the panel that opens customize the priority and return route. In this way you can force a specific node to be reached by controller (and also the opposite) using your custom defined route.
